modules/ROOT/pages/logging-in-mule.adoc

```diff
@@ -0,0 +1,289 @@
+    LOGGER.error("Failed to start asynchronous logger", ex);
+  }
+
+  @Override
+  public void handleOnShutdownException(Throwable ex) {
+    LOGGER.error("Failed to stop asynchronous logger", ex);
+  }
+}
+----
+
+There is a performance-reliability trade-off between asynchronous and synchronous logging. If the risk of losing log messages is a serious issue, then configure your loggers to be synchronous. You can have a mix of both synchronous and asynchronous logging.
+
+== Configuration Reloading
+
+Mule runtime uses Log4j 2 as the framework for logging. To add or change the monitoring interval at which Log4j 2 checks for changes, add the `monitorInterval` to the `log4j2.xml` configuration file:
+
+. In the Mule installation folder, open `conf/log4j2.xml`.
+. Add `<Configuration monitorInterval="15">` after the first line. For example:
++
+[source,xml,linenums]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration monitorInterval="15">
+    <Appenders>
+       <Console name="Console" target="SYSTEM_OUT">
+           <PatternLayout pattern="%-5p %d [%t] %c: %m%n"/>
+       </Console>
+    </Appenders>
+...
+----
++
+In this example, Log4j 2 checks for changes every 15 seconds.
++
+. Save the file, and restart the Mule server.
+
+For more information, see the Apache documentation on automatically configuring Log4j 2.
+
+== Debugging Projects that Use the HTTP Connector
+
+To debug projects that use the HTTP connector, you might want to make the logging more verbose than usual and track all of the behavior of both the Http-Listener and Http-Request connectors in your project. See xref:connectors::http/http-troubleshooting.adoc[HTTP Connector Troubleshooting Guide] for information.
+
+== Request and Response Logging for SOAP
+
+One common requirement during development is to be able to log the raw data of requests and responses for web services calls, especially SOAP calls. To handle this:
+
+//. In Anypoint Studio,
+
+. Open the `log4j2.xml` file in `src/main/resources` and add the HttpMessageLogger to the config:
++
+[source,xml,linenums]
+----
+<!-- HTTP is used by Mule for dispatching to web services -->
+<AsyncLogger name="org.mule.service.http.impl.service.HttpMessageLogger" level="DEBUG" />
+----
++
+. Save your project.
+
+== Troubleshooting Logging
+
+*I don't see any logging output*
+
+Set the `log4j2.xml` at the root of your classpath.
+
+*I reconfigured Log4j 2, but nothing happened*
+
+This happens because there is another `log4j2.xml` file on your classpath that is getting picked up before your modified one. To find out which configuration file Log4j 2 is using, add the following switch when starting Mule (or add it to the container startup script if you are embedding Mule):
+
+----
+-M-Dlog4j.debug=true
+----
+
+This parameter writes the Log4j 2 startup information, including the location of the configuration file being used, to `stdout`. You must remove that configuration file before your modified configuration can work.
```

Comment:
We called it a switch above, now a parameter.

Reply:
Minor: "being used" is passive, but it might be diff to turn that around.

Reply:
First comment addressed. Left the passive,